# แบบฝึกหัด 1

---

### 1.1

***เขียนโปรแกรมเพื่อเปลี่ยนค่าจากองศาเซลเซียสไปเป็นองศาฟาเรนไฮต์และเคลวิน ดังนี้***

F = (9/5)C + 32  
K = C + 273.15

รับข้อมูลองศาเซลเซียส

In [46]:
C = [39.85, 25, 32]

###
แปลงเป็นองศาฟาเรนไฮต์ให้ได้ผลลัพธ์เป็น list เก็บใน F

In [ ]:
# Your code
F = [(9 / 5) * c + 32 for c in C]
print(F)

[103.73, 77.0, 89.6]


ผลลัพธ์ที่ต้องการแสดง [103.73, 77.0, 89.6]

###
แปลงเป็นองศาฟาเคลวินให้ได้ผลลัพธ์เป็น list เก็บใน K

In [47]:
# Your code
K = [c + 273.15 for c in C]
print(K)

[313.0, 298.15, 305.15]


ผลลัพธ์ที่ต้องการแสดง [313.0, 298.15, 305.15]

###
แสดงการสรุปผล

In [ ]:
# Your code
for i in range(len(C)):
    print(f"{C[i]} °C = {F[i]} °F  = {K[i]} °K")

39.85 °C = 103.73 °F  = 313.0 °K
25 °C = 77.0 °F  = 298.15 °K
32 °C = 89.6 °F  = 305.15 °K


ผลที่ต้องการแสดง  
39.85 °C = 103.73 °F = 313.0 °K  
25 °C = 77.0 °F = 298.15 °K  
32 °C = 89.6 °F = 305.15 °K  

---

### 1.2

ให้นิสิตแสดงวิธีการแปลง list income (ซึ่งเปรียบเสมือนเป็นข้อมูลเงินเดือนของผู้ตอบแบบสอบถาม 20 ท่าน)  
ให้เป็น numpy array เก็บในตัวแปรที่ชื่อ income_number โดย array ที่แปลงได้ควรมีชนิด dtype เป็น float   
การแปลง string ให้เป็น float กำหนดให้ K = 1000 และ M = 1000000   
string "-" ให้แปลงเป็น np.nan (dtype ต้องเป็น float จึงจะใช้ np.nan ได้)

\* โจทย์ข้อนี้สามารถทำได้หลายวิธี  
หากใช้ numpy นิสิตอาจต้องหาข้อมูลเพิ่ม เช่น การใช้ np.char.replace หรือฟังก์ชัน eval

In [3]:
import numpy as np 

In [48]:
income = ["20K", "3M", "12K", "0.04M", "0.023K", "-", "16K", "740K", "15K", "40K",
         "-", "17K", "-", "120K", "56K", "-", "42K", "2M", "0.14M", "60K"]

In [ ]:
eval("2*1000")

2000

In [49]:
# นิสิตแสดงวิธีการด้วยการเขียนโค้ด
income = np.char.replace(income, "K", "*1000")
income = np.char.replace(income, "M", "*1000000")
income_number = np.array([eval(x) if x != "-" else np.nan for x in income])

##
วิธีการของนิสิตควรได้ผลลัพธ์ดังที่ระบุเมื่อรัน cell ต่อไปนี้

In [50]:
print(income_number)

[2.0e+04 3.0e+06 1.2e+04 4.0e+04 2.3e+01     nan 1.6e+04 7.4e+05 1.5e+04
 4.0e+04     nan 1.7e+04     nan 1.2e+05 5.6e+04     nan 4.2e+04 2.0e+06
 1.4e+05 6.0e+04]


ผลลัพธ์ที่ควรแสดง: array([2.0e+04, 3.0e+06, 1.2e+04, 4.0e+04, 2.3e+01,     nan, 1.6e+04,
       7.4e+05, 1.5e+04, 4.0e+04,     nan, 1.7e+04,     nan, 1.2e+05,
       5.6e+04,     nan, 4.2e+04, 2.0e+06, 1.4e+05, 6.0e+04])

In [31]:
type(income_number)

numpy.ndarray

ผลลัพธ์ที่ควรแสดง: numpy.ndarray

In [30]:
income_number.dtype

dtype('float64')

ผลลัพธ์ที่ควรแสดง: dtype('float64')

In [43]:
np.nanmean(income_number)

383625.0126875

ผลลัพธ์ที่ควรแสดง: 394876.4375

In [51]:
np.nanmedian(income_number)

41000.0

ผลลัพธ์ที่ควรแสดง: 41000.0

---